From 88ab6bc2dc81a09b8d910b57dc8739738528ae2e Mon Sep 17 00:00:00 2001
From: Karl Wette <karl.wette@ligo.org>
Date: Thu, 28 Aug 2025 12:34:12 +1000
Subject: [PATCH] lalpulsar/bin/HWInjection/hwinject.c: check errno only if
 popen() fails

- from https://man7.org/linux/man-pages/man3/errno.3.html:
    The value in errno is significant only when the return value of
    the call indicated an error (i.e., -1 from most system calls; -1
    or NULL from most library functions); a function that succeeds is
    allowed to change errno.
- closes lscsoft/lalsuite#808
---
 lalpulsar/bin/HWInjection/hwinject.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lalpulsar/bin/HWInjection/hwinject.c b/lalpulsar/bin/HWInjection/hwinject.c
index 54246dcd48..391b621287 100644
--- a/lalpulsar/bin/HWInjection/hwinject.c
+++ b/lalpulsar/bin/HWInjection/hwinject.c
@@ -851,7 +851,7 @@ int main( int argc, char *argv[] )
       printf( "[%02d] %s\n", i, command );
     } else {
       errno = 0;
-      if ( !( fp[i] = popen( command, "r" ) ) || errno ) {
+      if ( !( fp[i] = popen( command, "r" ) ) ) {
         syserror( 1, "Unable to popen(3) %s\n", command );
         exit( 1 );
       }
-- 
2.47.3

